### PR TITLE
refactor dialog + disable scrim prop

### PR DESCRIPTION
**Pull request by lukeac123**
- Add disableScrim prop + examples 
- Fix useFloatingUI focus manager implementation

### CODE REVIEW

File: packages/lab/src/dialog/Dialog.tsx [hunk 1]
```diff
@@ -15,19 +15,30 @@ import {
...
-  useId,
 } from "@salt-ds/core";
 import { useWindow } from "@salt-ds/window";
 import { useComponentCssInjection } from "@salt-ds/styles";
 import dialogCss from "./Dialog.css";
 import { DialogContext } from "./DialogContext";
 
+interface ConditionalScrimWrapperProps extends React.PropsWithChildren {
+  condition: boolean;
+}
```
Comment (joshwooding) [inline packages/lab/src/dialog/Dialog.tsx:33]:
```suggestion
interface ConditionalScrimWrapperProps {
  children?: ReactNode
  condition: boolean;
}
```

Comment (joshwooding) [reply packages/lab/src/dialog/Dialog.tsx:33]: More of a stylistic thing

File: packages/lab/src/dialog/Dialog.tsx [hunk 2]
```diff
@@ -15,19 +15,30 @@ import {
...
+
+export const ConditionalScrimWrapper = ({
+  condition,
+  children,
+}: ConditionalScrimWrapperProps) => {
+  return condition ? <Scrim fixed> {children} </Scrim> : <>{children} </>;
```
Comment (joshwooding) [inline packages/lab/src/dialog/Dialog.tsx:39]:
```suggestion
  return condition ? <Scrim fixed>{children}</Scrim> : <>{children} </>;
```

File: packages/lab/src/dialog/Dialog.tsx [hunk 3]
```diff
@@ -119,39 +135,39 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
...
-            {...rest}
-          >
-            {children}
-          </FloatingComponent>
-        </Scrim>
-      )}
+      <ConditionalScrimWrapper condition={open && !disableScrim}>
+        <FloatingComponent
+          open={showComponent}
+          role={role}
```
Comment (joshwooding) [inline packages/lab/src/dialog/Dialog.tsx:141]: role usage here can be simplified by hardcoding it here: role="dialog" and allowing role to be spread below by removing it from being destructured. Also we need to look at useRole again. I didn't realise floating-ui will change things depending on what value you provide it so need to double check it doesn't do anything special with dialog.

Comment (lukeac123) [reply packages/lab/src/dialog/Dialog.tsx:141]: when role='alert dialog' is passed will both 'dialog' and 'alert dialog' roles then not be attached to the same element ?
This no longer uses the useRole hook from floating-ui, you thinking we should be added back in ?

File: packages/lab/stories/dialog/dialog.stories.tsx [hunk 4]
```diff
@@ -233,12 +238,12 @@ export const MandatoryAction: StoryFn<typeof Dialog> = ({
       <Dialog
         size={"small"}
         status={"error"}
+        id={"Mandatory Action"}
```
Comment (Fercas123) [inline packages/lab/stories/dialog/dialog.stories.tsx:241]: nit
```suggestion
        id="Mandatory Action"
```

File: packages/lab/src/dialog/Dialog.tsx [hunk 5]
```diff
@@ -119,39 +132,38 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
...
-            {...rest}
-          >
-            {children}
-          </FloatingComponent>
-        </Scrim>
-      )}
+      <ConditionalScrimWrapper condition={open && !disableScrim}>
+        <FloatingComponent
+          open={showComponent}
+          role={"dialog"}
```
Comment (Fercas123) [inline packages/lab/src/dialog/Dialog.tsx:138]: nit
```suggestion
          role="dialog"
```

**Review by Fercas123 — DISMISSED** [review]

**Review by Fercas123 — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**